db/config.cc

```diff
@@ -953,6 +953,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , shutdown_announce_in_ms(this, "shutdown_announce_in_ms", value_status::Used, 2 * 1000, "Time a node waits after sending gossip shutdown message in milliseconds. Same as -Dcassandra.shutdown_announce_in_ms in cassandra.")
     , developer_mode(this, "developer_mode", value_status::Used, DEVELOPER_MODE_DEFAULT, "Relax environment checks. Setting to true can reduce performance and reliability significantly.")
     , skip_wait_for_gossip_to_settle(this, "skip_wait_for_gossip_to_settle", value_status::Used, -1, "An integer to configure the wait for gossip to settle. -1: wait normally, 0: do not wait at all, n: wait for at most n polls. Same as -Dcassandra.skip_wait_for_gossip_to_settle in cassandra.")
+    , gossip_endpoint_expiration_time(this, "gossip_endpoint_expiration_time", value_status::Used, 24 * 60 * 60 * 3, "Amount of time (in seconds) an unavailable node should be kept in gossip.")
```

Comment:
If a node is not removed from the cluster by nodetool decommission or nodetool removenode, the unavailable  node (e.g., a node is down and network connection to it is down) will be in gossip forever.  So the description above is not very accurate. 

The time is really about when gossip will remove gossip information for nodes that have been removed. 

I do not know why you want to change the default 3 days time. If you want to make it longer or shorter. 

I guess you probably want to make it shorter to have get rid of the removed nodes faster. This is not very useful in practice.  It is also not safe to make it short because other nodes might miss the LEFT status of the node if you remove it too fast. 

BTW, we are in the direction of making less dependency on gossip as possible in the system.

Reply:
I see, I got the impression that when a nodes gets the down status, after the aforementioned 72 hours it gets removed from the cluster automatically (by removal from gossip).
Do you have an idea of where it should be implemented to remove these unavailable nodes after the set duration?

Reply:
It is recommended to perform replace node operation[1] when a node is down.  In case you do not have a spare node to replace, you can run removenode ops[2] to remove the node from the cluster. 


[1] https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/cluster-management/replace-dead-node.html

[2] https://enterprise.docs.scylladb.com/stable/operating-scylla/nodetool-commands/removenode.html